src/api/db/migrate/20140624101042_add_package_tracking.rb

```diff
@@ -7,7 +7,7 @@ def up
 
       t.integer :release_package_id
 
-      t.string :binary_name, null: false
+      t.string :binary_name, null: false, charset: 'utf8'
```

Comment:
Please don't alter old migrations. If you need to change the charset (before switching to utf8mb) write a migration in between.

Reply:
> the rake task to replay history is using the same database.yml that we need for other tests

Then change that

Reply:
Thought about that - but it will only help for this very moment. All future migrations you'd expect to work on utf8mb4

Reply:
>  All future migrations you'd expect to work on utf8mb4

Hm isn't that clear if we switch to it? I don't get it...

Reply:
the dev:lint:db rake task replays all migration history in one go. Right now (master) this is done with a database.yml that contains since forever encoding: utf8 and all is good - unless you want to write real unicode.

now we want to switch, so we switch the database table format to barracuda to erase the index limit and change some columns. Perfect migration, no harm done.

*BUT*: to actually use this feature you also have to switch your DB connection to use the utf8mb4 encoding and this is done in database.yml. But if we change database.yml, all *old* add_column calls suddenly find themselves caught in new the index limit not matching a varchar(255). My solution to that problem is: make the old add_column calls use the old encoding, so we don't change history.

What you propose is *not* changing database.yml for migration testing, so they don't break and we don't have to touch them. But then you have to ask yourself: why are we doing migration testing - now and in the future. We do that so that users can run the old migrations no mater how our new code looks like. And future migrations we'd write with utf8mb4 requirement, so we wouldn't be able to test *them* correctly with an old database.yml.